x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/plugin/ScoreFunctionIT.java

```diff
@@ -0,0 +1,413 @@
+            assertValues(
+                resp.values(),
+                List.of(
+                    List.of(1, 1.4274532794952393, 0.2708943784236908),
+                    List.of(2, 0.2708943784236908, 0.2708943784236908),
+                    List.of(3, 0.2708943784236908, 0.2708943784236908),
+                    List.of(4, 0.19301524758338928, 0.19301524758338928),
+                    List.of(6, 1.1248724460601807, 0.21347221732139587)
+                )
+            );
+        }
+    }
+
+    public void testScoreBothWhereQstrORMatch() {
+        var query = """
+            FROM test METADATA _score
+            | WHERE qstr("brown") OR match(content, "fox")
+            | EVAL first_score = score(qstr("brown"))
+            | EVAL second_score = score(match(content, "fox"))
+            | KEEP id, _score, first_score, second_score
+            | SORT id
+            """;
+
+        try (var resp = run(query)) {
+            assertColumnNames(resp.columns(), List.of("id", "_score", "first_score", "second_score"));
+            assertColumnTypes(resp.columns(), List.of("integer", "double", "double", "double"));
+            assertValues(
+                resp.values(),
+                List.of(
+                    List.of(1, 1.4274532794952393, 0.2708943784236908, 1.156558871269226),
+                    List.of(2, 0.2708943784236908, 0.2708943784236908, 0.0),
+                    List.of(3, 0.2708943784236908, 0.2708943784236908, 0.0),
+                    List.of(4, 0.19301524758338928, 0.19301524758338928, 0.0),
+                    List.of(6, 1.1248724460601807, 0.21347221732139587, 0.9114001989364624)
+                )
+            );
+        }
+    }
+
+    public void testSimpleScoreAlone() {
+        var query = """
+            FROM test METADATA _score
+            | EVAL first_score = score(match(content, "brown"))
+            | KEEP id, _score, first_score
+            | SORT id
+            """;
+
+        try (var resp = run(query)) {
+            assertColumnNames(resp.columns(), List.of("id", "_score", "first_score"));
+            assertColumnTypes(resp.columns(), List.of("integer", "double", "double"));
+            assertValues(
+                resp.values(),
+                List.of(
+                    List.of(1, 0.0, 0.2708943784236908),
+                    List.of(2, 0.0, 0.2708943784236908),
+                    List.of(3, 0.0, 0.2708943784236908),
+                    List.of(4, 0.0, 0.19301524758338928),
+                    List.of(5, 0.0, 0.0),
+                    List.of(6, 0.0, 0.21347221732139587)
+                )
+            );
+        }
+    }
+
+    private void createAndPopulateIndex() {
+        var indexName = "test";
+        var client = client().admin().indices();
+        var CreateRequest = client.prepareCreate(indexName)
+            .setSettings(Settings.builder().put("index.number_of_shards", 1))
```

Comment:
I know it cerated inconsistency on the actual values of the scores but I think it could be worth to run some test cases using a multiple shard setup at least to check that it is executed correctly (and maybe that all the expected hits are returned even with different scores or in a different order).

WDYT?

Reply:
I will do that in a separate PR, as that also affects the `ScoringIT`.